src/rpdk/core/build_image.py

```diff
@@ -0,0 +1,64 @@
+"""This sub command will build an image for the executable
+
+Projects can be created via the 'init' sub command.
+"""
+import logging
+import os
+
+import docker
+
+from .project import Project
+
+LOG = logging.getLogger(__name__)
+VALID_IMAGE_RUNTIMES = [
+    "java8",
```

Comment:
Could we delegate the docker image and runtime configuration to each plugin?

Reply:
Could you clarify what you mean by delegate the image and runtime configuration to the plugin? Are you saying create the dockerfile as part of the plugin? So, for instance, that it would be created when running cfn generate?

For me, it makes more sense to have this logic as part of the cloudformation cli. The plugins shouldn't really care how the executables are being consumed. Furthermore, I thought that putting everything in the plugin would require more effort for existing projects to onboard this feature. We would probably need to add the executable name as an additional field in .rpdk_config which looked like was only generated once during project initialization.

Reply:
Talked offline. The plugin can generate the image settings in a separate method like:

```
image_config = project._plugin.generate_image_config()
```

Made these changes